network/router.go

```diff
@@ -97,7 +97,7 @@ func (r *Router) Unpause() {
 
 // Start the listening routine of the underlying Host. This is a
 // blocking call until r.Stop() is called.
-func (r *Router) Start() {
+func (r *Router) Start(isPeerValid func(ServerIdentityID) bool) {
```

Comment:
This would break the public API of `Router`, and I don't see an imperative reason to do so.
Can you rather keep the `isPeerValid` as an internal field that can be overwritten by the tests?
Or use a `SetPeerValidation` method to change the default peer validation.
Or add a `StartWithPeerValidation`, then have `Start` call `StartWithPeerValidation(r.defaultPeerValidation)`?

Reply:
I found it cleaner this way, but indeed it breaks `Router`'s public API.
I will change it.